app/seller/item-form.tsx

```diff
@@ -0,0 +1,163 @@
+import { View, Text, TextInput, TouchableOpacity, Image } from "react-native";
+import { Control, Controller, UseFieldArrayRemove } from "react-hook-form";
+import { PostFormType } from "@/types/post";
+
+
+interface ItemFormProps {
+    control: Control<PostFormType>;
+    index: number;
+    field: Record<string, any>;
+    errors: any;
+    remove: UseFieldArrayRemove;
+    images: Record<string, string>;
+    setImages: React.Dispatch<React.SetStateAction<Record<string, string>>>;
+    openImageLibrary: (fieldId: string) => Promise<void>;
+    openCamera: (fieldId: string) => Promise<void>;
+    fieldsLength: number;
+}
+
+function ItemForm({
+    control,
+    index,
+    field,
+    errors,
+    remove,
+    images,
+    setImages,
+    openImageLibrary,
+    openCamera,
+    fieldsLength,
+}: ItemFormProps) {
```

Comment:
lets use this convention for all components that has props. 

```tsx
function ItemForm(props: ItemFormProps) {
    const {
        control,
        index,
        field,
        errors,
        remove,
        images,
        setImages,
        openImageLibrary,
        openCamera,
        fieldsLength,
    } = props;
}
```

Lets not deconstruct the props in the parenthesis. We just name the parameter as `props` and we deconstruct inside the component.